packages/core/src/utils.ts

```diff
@@ -92,6 +92,17 @@ export class Utils {
       if (!linkedCommit) throw new Error(`No commit found for CID ${commit.link.toString()}`)
       commitData.commit = linkedCommit
       commitData.envelope = commit
+
+      if (commit.signatures && commit.signatures.length > 0) {
+        const protectedHeader = commit.signatures[0].protected
```

Comment:
nit: can you pull this out into an `extractCapability` helper function please?

Reply:
Would also be good if that function could catch an error loading the capability from IPFS and re-throw a more descriptive error message indicating that the CID that failed to load was that of the capability